runtime/logging_test.go

```diff
@@ -45,3 +50,167 @@ func TestDropInputParam(t *testing.T) {
+	e.Level = logrus.InfoLevel
+
+	out, err := fmtr.Format(e)
+	if err != nil {
+		t.Fatalf("Unexpected error formatting log entry: %s", err.Error())
+	}
+
+	actualStr := string(out)
+
+	expectedLvl := strings.ToUpper(e.Level.String())
+	if !strings.Contains(actualStr, expectedLvl) {
+		t.Errorf("Expected log message to have level %s:\n%s", expectedLvl, actualStr)
+	}
+
+	if !strings.Contains(actualStr, "test") {
+		t.Errorf("Expected log message to have the entry message '%s':\n%s", "test", actualStr)
+	}
+
+	if !strings.Contains(actualStr, "number = 5") {
+		t.Errorf("Expected to have the number field in message")
+	}
+
+	if !strings.Contains(actualStr, "string = \"field_string\"") {
+		t.Errorf("Expected to have the string field in message")
+	}
+
+	if !strings.Contains(actualStr, "nil = null") {
+		t.Errorf("Expected to have the nil field in message")
+	}
+
+	if !strings.Contains(actualStr, "error = \"field_error\"") {
+		t.Errorf("Expected to have the nil field in message")
+	}
+}
+
+func TestPrettyFormatterMultilineStringFields(t *testing.T) {
+	fmtr := prettyFormatter{}
+
+	mlStr := `
+a
+bcdef
+g
+
+{ heijk }	
+
+`
+
+	e := logrus.WithFields(logrus.Fields{
+		"multi_line": mlStr,
+	})
+
+	e.Message = "test"
+	e.Level = logrus.InfoLevel
+
+	out, err := fmtr.Format(e)
+	if err != nil {
+		t.Fatalf("Unexpected error formatting log entry: %s", err.Error())
+	}
+
+	actualStr := string(out)
+
+	expectedLvl := strings.ToUpper(e.Level.String())
+	if !strings.Contains(actualStr, expectedLvl) {
+		t.Errorf("Expected log message to have level %s:\n%s", expectedLvl, actualStr)
+	}
+
+	if !strings.Contains(actualStr, "test") {
+		t.Errorf("Expected log message to have the entry message '%s':\n%s", "test", actualStr)
+	}
+
+	for _, line := range strings.Split(mlStr, "\n") {
+		// The lines will get prefixed with some padding but should always
+		// still have their real newlines, and not be encoded.
+		expectedStr := line + "\n"
+		if !strings.Contains(actualStr, expectedStr) {
+			t.Errorf("Expected to find line in message:\n\n%s\n\nactual:\n\n%s\n", expectedStr, actualStr)
+		}
+	}
+}
+
+func TestPrettyFormatterMultilineJSONFields(t *testing.T) {
+	fmtr := prettyFormatter{}
+
+	obj := map[string]interface{}{
+		"a": 123,
+		"b": nil,
+		"d": "abc",
+		"e": map[string]interface{}{
+			"test": []string{
+				"aa",
+				"bb",
+				"cc",
+			},
+		},
+	}
+
+	e := logrus.WithFields(logrus.Fields{
+		"json_string": obj,
+	})
+
+	e.Message = "test"
+	e.Level = logrus.InfoLevel
+
+	out, err := fmtr.Format(e)
+	if err != nil {
+		t.Fatalf("Unexpected error formatting log entry: %s", err.Error())
+	}
+
+	actualStr := string(out)
+
+	expectedLvl := strings.ToUpper(e.Level.String())
+	if !strings.Contains(actualStr, expectedLvl) {
+		t.Errorf("Expected log message to have level %s:\n%s", expectedLvl, actualStr)
+	}
+
+	if !strings.Contains(actualStr, "test") {
+		t.Errorf("Expected log message to have the entry message '%s':\n%s", "test", actualStr)
```

Comment:
Nit: I think `'%s' ` is the same as `%q`

Reply:
If I understand correctly `%q` does character escaping too, which I don't like to use in the test error strings since we want to show the exact string we are comparing with. I guess in this case its just `"test"` so it doesn't really matter, doesn't even need to be a parameter 😃